src/python/platforms/fuchsia/device.py

```diff
@@ -16,6 +16,142 @@
 # TODO(mbarbella): Re-enable this check once functions below are implemented.
 # pylint: disable=unused-argument
 
+import os
+
+from google_cloud_utils import gsutil
+from metrics import logs
+from platforms.fuchsia import errors
+from system import environment
+from system import new_process
+from system import shell
+
+
+def qemu_setup():
+  """ Sets up and runs a QEMU VM in the background.
+  Does not block the calling process.
+  Fuchsia fuzzers assume a QEMU VM is running; call this routine prior to
+  beginning Fuchsia fuzzing tasks.
+  This initialization routine assumes that the GCS bucket contains the
+  standard Fuchsia SDK, as well as:
+  * /qemu-for-fuchsia/*
+  * /.ssh/* """
+  # First download the Fuchsia resources locally.
+  fuchsia_resources_dir = initialize_resources_dir()
+
+  # Then, save paths for necessary commands later.
+  qemu_path = os.path.join(fuchsia_resources_dir, 'qemu-for-fuchsia', 'bin',
+                           'qemu-system-x86_64')
+  os.chmod(qemu_path, 0o550)
+  kernel_path = os.path.join(fuchsia_resources_dir, 'target', 'x64',
+                             'qemu-kernel.bin')
+  os.chmod(kernel_path, 0o644)
+  pkey_path = os.path.join(fuchsia_resources_dir, '.ssh', 'pkey')
+  os.chmod(pkey_path, 0o400)
+  sharefiles_path = os.path.join(fuchsia_resources_dir, 'qemu-for-fuchsia',
+                                 'share', 'qemu')
+  drive_path = os.path.join(fuchsia_resources_dir, 'target', 'x64', 'fvm.blk')
+  os.chmod(drive_path, 0o644)
+  fuchsia_zbi = os.path.join(fuchsia_resources_dir, 'target', 'x64',
+                             'fuchsia.zbi')
+  initrd_path = os.path.join(fuchsia_resources_dir, 'fuchsia-ssh.zbi')
+  os.chmod(initrd_path, 0o644)
+
+  # Perform some more initiailization steps.
+  extend_fvm(fuchsia_resources_dir, drive_path)
+  add_keys_to_zbi(fuchsia_resources_dir, initrd_path, fuchsia_zbi)
+
+  # TODO(flowerhack): Add a mechanism for choosing portnum dynamically.
+  portnum = '56339'
+
+  # yapf: disable
+  qemu_command = [
```

Comment:
Maybe just call this `qemu_args` and remove `qemu_path`?
That way we don't need to slice the list.